leo/package/src/package.rs

```diff
@@ -51,57 +51,36 @@ impl Package {
 
     /// Returns `true` if the package name is valid.
     ///
-    /// Package names must be lowercase and composed solely
-    /// of ASCII alphanumeric characters, and may be word-separated
-    /// by a single dash '-'.
+    /// Package names must be composed solely of ASCII alphanumeric characters,
+    /// and may be word-separated by a single underscore '_'.
```

Comment:
It also allows contiguous underscores (which is consistent with the definition of identifiers), so perhaps this comment should be slightly rephrased. Could say something like "Package names must identifiers, i.e. composed solely of ASCII alphanumeric characters and underscores, starting with a letter." (Note that "starting with a letter" implies non-empty.)